src/Simplify.elm

```diff
@@ -1554,55 +1557,13 @@ expressionVisitorHelp node context =
         --  APPLIED LAMBDA FUNCTIONS --
         -------------------------------
         Expression.Application ((Node _ (Expression.ParenthesizedExpression (Node lambdaRange (Expression.LambdaExpression lambda)))) :: firstArgument :: _) ->
```

Comment:
The fact that this is only reported on direct apply, not `|>` or `<|` avoids many intentional situations like
```elm
pieceBodies
    |> List.map .data
    |> (\pieces -> newPiecesThisRound ++ pieces)
    |> takeFormations
    |> toBodies
```
so this rule gets a :+1: from me

Reply:
Ah, I actually forgot about these. I guess I kind of want to support them to a certain extent (especially since this is not even resolving @lydell's original issue...) but I agree that we don't want to report an issue for the example you just showed.

Maybe the heuristic should be that if we pipe an expression to the anonymous function, then we only report it if the argument is not a pipeline itself.

```elm
-- Reported
pieceBodies
    |> (\pieces -> newPiecesThisRound ++ pieces)
    |> f

-- NOT reported
pieceBodies
    |> List.map .data
    |> (\pieces -> newPiecesThisRound ++ pieces)
```

And same thing for `<|`. What do you think?

Reply:
If this reports only `(\b -> b * 2) a` and not `a |> (\b -> b * 2)`, then it is a useful simplification, but it won’t solve the problem I reported. I don’t think I’ve ever encountered `(\b -> b * 2) a` in real code.

I do however have encountered `a |> (\b -> b * 2)`, in two types of situations:

1. Because the Simplify rule simplified a pipeline into that (like I mentioned in #111).
2. Because colleagues go into “pipeline mode” and forget that let-in exists. See below.

I personally _always_ dislike piping into a lambda, but I can agree on it in @lue-bird’s example with a longer pipeline. However, I don’t think it makes sense to _start_ a pipeline with a lambda.

```elm
bad1 =
    createFruitBasket seeds prices recommendations
        |> (\fruitBasket ->
                { fruits = fruitBasket.fruits
                , size = fruitBasket.sizes |> List.sum
                }
           )


good1 =
    let
        fruitBasket =
            createFruitBasket seeds prices recommendations
    in
    { fruits = fruitBasket.fruits
    , size = fruitBasket.sizes |> List.sum
    }


bad2 =
    createFruitBasket seeds prices recommendations
        |> (\fruitBasket -> fruitBasket.apples ++ fruitBasket.oranges)
        |> List.map viewFruit


good2 =
    let
        fruitBasket =
            createFruitBasket seeds prices recommendations
    in
    fruitBasket.apples
        ++ fruitBasket.oranges
        |> List.map viewFruit
```

Reply:
I think it *can* make sense to wrap in a lambda is something a bit complex, like an `if` condition.

```elm
bad1 =
    createFruitBasket seeds prices recommendations
        |> (\fruitBasket ->
              if condition then
                Just { fruits = fruitBasket.fruits
                        , size = fruitBasket.sizes |> List.sum
                        }
             else
                Nothing
           )
```

though I would personally always recommend using a `let` variable instead.

But I feel like my proposal follows @lydell's examples, so let's go with that. I'll probably do this in a separate PR though.

Reply:
You _might_ be able to make a case that more complex arguments are fine to be piped into a lambda
```elm
piecesExtractFromBodies
    { physicsBodies = model.physicsBodies
    , activeCombineStrategy = model.combineStrategy
    , countCombinedAsSeparate = True
    }
    |> (\pieces -> newPiecesThisRound ++ pieces)
    |> ...continues
```
My initial reaction was for this rule to only report with "simple" arguments (literals and function/value references) for pipes to be reported.
But now I think your suggestion will probably be a good rule and above style isn't all that great :+1: